BUILD.md

```diff
@@ -35,18 +35,18 @@ You can check that the corresponding `.jar` file is present in your local maven
 ## Deterministic build of Phoenix
 
 Phoenix supports deterministic builds on Linux OSs, this allows anyone to recreate from the sources the exact same APK that was published in the release page.
-The deterministic build uses a dockerized build environment and require you to have previously built (and published locally) the artifact for the `eclair-core`
-dependency, follow the instructions to build it.
+The deterministic build uses a self-contained dockerized environment, to run the build you must supply via docker build args the branch of eclair-core
+that is going to be used for the build. Each release of Phoenix is built against a specific revision of eclair-core, checkout the release notes to know
+which revision you need to use in the deterministic build.
 
 ### Prerequisites
 
 1. A linux machine running on x64 CPU.
 2. docker-ce installed
-3. Eclair-core published in your local maven repo, check out the instructions to build it.
 
 ### How to build phoenix deterministically
 
 1. Clone the phoenix project from https://github.com/ACINQ/phoenix
-3. Run `docker build -t phoenix_build .` to create the build environment
-4. Run `docker run --rm -v $HOME/.m2:/root/.m2 -v $(pwd):/home/ubuntu/phoenix/app/build -w /home/ubuntu/phoenix phoenix_build ./gradlew assemble`
-5. Built artifacts are in $(pwd)/outputs/apk/release
+3. Run `docker build -t phoenix_build . --build-arg ECLAIR_TAG=<GIT_TAG_OF_ECLAIR_CORE>` to create the build environment
```

Comment:
Where does `<GIT_TAG_OF_ECLAIR_CORE>` come from? How about you merge Could you merge #43 instead?

Reply:
I'm not a big fan of git submodules but i reckon in this scenario they could prove useful because we can pinpoint the tag of eclair-core from the phoenix repo, however we usually think of eclair-core as a separate library (is used in 1+ projects) and we'd like to keep it that way. Since every release of Phoenix is build against a particular tag of eclair-core `GIT_TAG_OF_ECLAIR_CORE` is the tag that you need to pass to the build, the actual tag should be published in the release notes. 

Note that with this new multilstage build completely removes the need to build external dependencies manually.